llm_demo/orchestrator/langchain_tools/tools.py

```diff
@@ -83,15 +85,18 @@ async def search_airports(country: str, city: str, name: str):
 
         num = 2
         response_json = await response.json()
-        if len(response_json) < 1:
+        if response_json.get("trace"):
+            tool_trace.add_message(response_json.get("trace"))
```

Comment:
```suggestion
       trace = response_json.get("trace")
        if trace:
            tool_trace.add_message(trace)
```

Reply:
Do we need to validate trace is in the right format somewhere?

Reply:
"trace" are build in retrieval service/app/routes.py (src code is in: retrieval_service/helpers/helpers.py) before sending it back to the UI.